pkg/bootstrap/platform/gcp.go

```diff
@@ -104,11 +110,29 @@ var (
 		return fmt.Sprintf("https://container.googleapis.com/v1/projects/%s/locations/%s/clusters/%s",
 			projectID, clusterLocation, clusterName), nil
 	}
+
+	isMetadataEndpointAccessibleFn = func() bool {
+		envOnce.Do(func() {
```

Comment:
See comment above, we may be able to avoid a `Once` if we store it in gcpEnv? Or we can have it still in `Once` to lazy load it, but scoped to gcpEnv instead of global?

Reply:
That's a great idea. I updated the code according to your suggestion